gcs/indexer.py

```diff
@@ -39,14 +41,53 @@ def parse_args():
     return parser.parse_args()
 
 
+def index_gcs_path(gcs_path):
+    # Input gcs_path looks like
+    # gs://genomics-public-data/platinum-genomes/bam/PRIMARY_KEY_
+
+    logger.info('Processing %s.' % gcs_path)
+
+    trimmed_gcs_path = gcs_path.replace('gs://', '')
+
+    # bucket_str looks like genomics-public-data
+    bucket_str = trimmed_gcs_path.split('/')[0]
+
+    prefix = trimmed_gcs_path.split('/', 1)[1]
+    # prefix looks like platinum-genomes/bam/
+    prefix = prefix[:prefix.index('PRIMARY_KEY')]
+
+    logger.info('Retrieving objects from bucket %s with prefix %s.' %
+                (bucket_str, prefix))
+    bucket = storage.Client(project=None).bucket(bucket_str)
+    objects = bucket.list_blobs(prefix=prefix)
+
+    regex = re.compile(gcs_path.replace('PRIMARY_KEY', '(\w+)'))
+
+    for obj in objects:
+        obj_path = 'gs://%s/%s' % (bucket_str, obj.name)
+        match = re.match(regex, obj_path)
+        if match:
+            primary_key = match.group(1)
+            print(
+                'Identified primary key %s from %s' % (primary_key, obj_path))
+        else:
+            raise ValueError('Could not find primary key in %s' % obj_path)
+
+
 def main():
     args = parse_args()
 
+    # Read dataset config files
     index_name = indexer_util.get_index_name(args.dataset_config_dir)
+    gcs_json = os.path.join(args.dataset_config_dir, 'gcs.json')
```

Comment:
`gcs_json` sounds like a JSON object or a string containing json. `gcs_config_path` would be clearer

Reply:
Done